hexapod/models.py

```diff
@@ -117,43 +117,43 @@ def __init__(self, dimensions):
         self._init_local_frame()
 
     def update(self, poses, assume_ground_targets=True):
-        self.body_rotation_frame = None
-        might_twist = find_if_might_twist(self, poses)
-        old_contacts = deepcopy(self.ground_contacts)
-
-        # Update leg poses
-        for pose in poses.values():
-            i = pose["id"]
-            if(poses_within_range(pose)):
+        if(poses_within_range(poses)):
```

Comment:
I prefer exiting early. as the zen of python says, flat is better than nested. 

IE

```
if poses_within_range(poses): 
    return

// do what you want to do here